### ISSUE TITLE
A3  crash with scan_mode Sensitivity

### ISSUE THREAD

**Issue by winf** [robopeak/rplidar_ros#79]
![crash](https://user-images.githubusercontent.com/1445783/47842048-894cb880-ddf6-11e8-8691-71ebbf3ee22f.PNG)
However ，scan_mode with stability is working.

**Comment by winf** [comment]
disable angle_compensate could resolve this

**Comment by winf** [comment]
maybe “ __attribute__((packed)) ” in rplidar_cmd.h #line204  cause this behave.
https://stackoverflow.com/questions/8568432/is-gccs-attribute-packed-pragma-pack-unsafe

**Comment by ramonvp** [comment]
Hi! Any news on this? I am also experiencing the same crash. It does not happen if you modify the launch file and use "Standard" as scan_mode instead of "Sensitivity"

**Comment by niralber** [comment]
same problem. disable angle_compensate solves the problem. but I don't know why and assume the accuracy is affected by this...

**Comment by cjmahieu** [comment]
I found that both the Standard and Express mode work. However I would like the Sensitivity mode to work as well..

Disabling angle compensate prevents the crash from happening but makes the scan data unusable as shown below:

![image](https://user-images.githubusercontent.com/46316517/53480321-42f12880-3a7b-11e9-8f59-c60d26af0d19.png)

(With express / standard option it looks as follows:)

![image](https://user-images.githubusercontent.com/46316517/53480409-70d66d00-3a7b-11e9-9c47-6e68a9d02eb8.png)

EDIT:

What i reported above was using the package installed via aptitute (i.e. ros-kinetic-rplidar-ros).
Using the version from this git repositoy I was able to run at 'Sensitivity' level if I turned of angle compensation.

Proof below:
![image](https://user-images.githubusercontent.com/46316517/53482739-36bb9a00-3a80-11e9-925b-9d5a64161a1c.png)

**Comment by ameykasar** [comment]
Had the same problem. Editing the scan mode in launch file (as mentioned above) helped.
The problem went away completely when I switched to another USB cable. Digging more I found out the cable was not capable of providing enough stable power for the sensor.
Observe the laser does not light up when you get the error as the power is not sufficient.

TLDR: CHANGE USB CABLE/ CHECK POWER DELIVERED TO THE SENSOR

**Comment by deb0ch** [comment]
Looks a lot like https://github.com/Slamtec/rplidar_ros/issues/1 and fixed by https://github.com/Slamtec/rplidar_ros/pull/10 (although I'm not comfortable with the arbitrary number chosen to extend the buffer).

I suspect this repo is deprecated now that the Slamtec one is there so this is FYI
